src/vm/costs/mod.rs

```diff
@@ -409,6 +457,26 @@ impl CostTracker for LimitedCostTracker {
             self.memory = 0;
         }
     }
+    fn short_circuit_contract_call(
+        &mut self,
+        contract: &QualifiedContractIdentifier,
+        function: &ClarityName,
+        input: &[u64],
+    ) -> Result<bool> {
+        if self.free {
+            // if we're already free, no need to worry about short circuiting contract-calls
+            return Ok(false);
+        }
+        // grr, if HashMap::get didn't require Borrow, we wouldn't need this cloning.
+        let lookup_key = (contract.clone(), function.clone());
+        if let Some(cost_function) = self.contract_call_circuits.get(&lookup_key).cloned() {
+            let input_size = input.iter().fold(0, |agg, cur| agg + cur);
```

Comment:
I guess I don't understand what `input` is here.  Is input just a list of instance characteristics?  If so, then why is _summing_ them the right thing to do?  Consider matrix multiplication as a counterexample -- the runtime would be the _product_ of the instance characteristics.  Or, am I misunderstanding what's going on here?

Reply:
That's the size of each input argument. Ultimately, this is going to change to just be multiple arguments, and then the defined cost function can choose how to combine them.

Reply:
Gotcha.  Is there an issue open for this?  If not, can you open one?

Reply:
Opened: https://github.com/blockstack/stacks-blockchain/issues/2075

Reply:
We've actually already got this issue: https://github.com/blockstack/stacks-blockchain/issues/2047